pkgs/applications/video/obs-studio/v4l2sink.nix

```diff
@@ -1,42 +1,49 @@
-{ stdenv, fetchFromGitHub
-, cmake, pkgconfig, wrapQtAppsHook
-, obs-studio }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, qt5
+, obs-studio
+}:
 
-stdenv.mkDerivation {
-  pname = "obs-v4l2sink-unstable";
-  version = "20181012";
+stdenv.mkDerivation rec {
+  pname = "obs-v4l2sink";
+  version = "0.1.0";
 
   src = fetchFromGitHub {
     owner = "CatxFish";
     repo = "obs-v4l2sink";
-    rev = "1ec3c8ada0e1040d867ce567f177be55cd278378";
-    sha256 = "03ah91cm1qz26k90mfx51l0d598i9bcmw39lkikjs1msm4c9dfxx";
+    rev = version;
+    sha256 = "0l4lavaywih5lzwgxcbnvdrxhpvkrmh56li06s3aryikngxwsk3z";
   };
 
-  nativeBuildInputs = [ cmake pkgconfig wrapQtAppsHook ];
-  buildInputs = [ obs-studio ];
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ qt5.qtbase.dev obs-studio ];
 
   patches = [
-    ./0001-find-ObsPluginHelpers.cmake-in-the-obs-src.patch
+    ./fix-v4l2device_close-segfault.patch
   ];
 
-  cmakeFlags = [
-    "-DLIBOBS_INCLUDE_DIR=${obs-studio}/include/obs"
-    "-DLIBOBS_LIBRARIES=${obs-studio}/lib"
-    "-DCMAKE_CXX_FLAGS=-I${obs-studio.src}/UI/obs-frontend-api"
-    "-DOBS_SRC=${obs-studio.src}"
+  cmakeFlags = with lib; [
+    "-DLIBOBS_INCLUDE_DIR=${obs-studio.src}/libobs"
+    "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}"
   ];
 
-  installPhase = ''
-    mkdir -p $out/share/obs/obs-plugins/v4l2sink/bin/64bit
-    cp ./v4l2sink.so $out/share/obs/obs-plugins/v4l2sink/bin/64bit/
+  pluginPath = {
+    i686-linux = "bin/32bit";
+    x86_64-linux = "bin/64bit";
+  }.${stdenv.targetPlatform.system} or (throw "Unsupported system: ${stdenv.targetPlatform.system}");
+
+  postInstall = ''
+    mkdir -p $out/share/obs/obs-plugins/v4l2sink/$pluginPath
+    ln -s $out/lib/obs-plugins/v4l2sink.so $out/share/obs/obs-plugins/v4l2sink/$pluginPath
```

Comment:
Why is this needed? Could you add a comment?

Reply:
obs-studio seems to expects a specific directory structure for plugins, otherwise it won't find it.

Reply:
Is that upstream thing or our packaging thing. If the former, why does the project not already do that?

Either way, the comment should be part of the expression.

Reply:
It's an upstream thing. It looks for plugins in `$XDG_CONFIG_HOME/obs-studio/plugins` and `/usr/share/obs/obs-plugins`